diff --git a/Sources/Sentry/SentryCrashReportConverter.m b/Sources/Sentry/SentryCrashReportConverter.m
index d6672fad..65a3ae51 100644
--- a/Sources/Sentry/SentryCrashReportConverter.m
+++ b/Sources/Sentry/SentryCrashReportConverter.m
@@ -264,6 +264,7 @@ - (SentryFrame *)stackFrameAtIndex:(NSInteger)frameIndex inThreadIndex:(NSIntege
     frame.instructionAddress = sentry_formatHexAddress(frameDictionary[@"instruction_addr"]);
     frame.imageAddress = sentry_formatHexAddress(binaryImage[@"image_addr"]);
     frame.package = binaryImage[@"name"];
+    frame.module = [binaryImage[@"name"] lastPathComponent];
     BOOL isInApp = [self.inAppLogic isInApp:binaryImage[@"name"]];
     frame.inApp = @(isInApp);
     if (frameDictionary[@"symbol_name"]) {
diff --git a/Sources/Sentry/SentryCrashStackEntryMapper.m b/Sources/Sentry/SentryCrashStackEntryMapper.m
index 4937146f..c92bfa33 100644
--- a/Sources/Sentry/SentryCrashStackEntryMapper.m
+++ b/Sources/Sentry/SentryCrashStackEntryMapper.m
@@ -45,6 +45,7 @@ - (SentryFrame *)sentryCrashStackEntryToSentryFrame:(SentryCrashStackEntry)stack
         NSString *imageName = [NSString stringWithCString:stackEntry.imageName
                                                  encoding:NSUTF8StringEncoding];
         frame.package = imageName;
+        frame.module = [imageName lastPathComponent];
         frame.inApp = @([self.inAppLogic isInApp:imageName]);
     }
 
diff --git a/Sources/Sentry/SentryMetricKitIntegration.m b/Sources/Sentry/SentryMetricKitIntegration.m
index df637276..3d2b67ec 100644
--- a/Sources/Sentry/SentryMetricKitIntegration.m
+++ b/Sources/Sentry/SentryMetricKitIntegration.m
@@ -271,6 +271,7 @@ - (SentryStacktrace *)convertMXFramesToSentryStacktrace:(NSEnumerator<SentryMXFr
     for (SentryMXFrame *mxFrame in mxFrames) {
         SentryFrame *frame = [[SentryFrame alloc] init];
         frame.package = mxFrame.binaryName;
+        frame.module = [mxFrame.binaryName lastPathComponent];
         frame.instructionAddress = sentry_formatHexAddress(@(mxFrame.address));
         NSNumber *imageAddress = @(mxFrame.address - mxFrame.offsetIntoBinaryTextSegment);
         frame.imageAddress = sentry_formatHexAddress(imageAddress);
