diff --git a/Podfile b/Podfile
index d2d83b14..6491abf8 100644
--- a/Podfile
+++ b/Podfile
@@ -30,7 +30,7 @@ pod 'ObjectMapper', git: 'https://github.com/tristanhimmelman/ObjectMapper.git',
 pod 'PromiseKit'
 
 pod 'RealmSwift'
-pod 'Sentry'
+pod 'Sentry', :git => 'https://github.com/getsentry/sentry-cocoa.git', :commit => '__GITHUB_REVISION_PLACEHOLDER__'
 pod 'UIColor_Hex_Swift'
 pod 'Version'
 pod 'XCGLogger'
