diff --git a/Podfile b/Podfile
index b9205a9f..3156735b 100644
--- a/Podfile
+++ b/Podfile
@@ -9,6 +9,7 @@ def shared_pods
   pod 'CocoaHTTPServer', :git => 'https://github.com/fkuehne/CocoaHTTPServer.git' # has our fixes
   pod 'xmlrpc', :git => 'https://github.com/fkuehne/xmlrpc.git', :commit => '3f8ce3a8' #fix case-sensitive FS
   pod 'AFNetworking', '~>4.0'
+  pod 'Sentry', :git => 'https://github.com/getsentry/sentry-cocoa', :commit => '__GITHUB_REVISION_PLACEHOLDER__'
 
   # debug
   pod 'SwiftLint', '~> 0.47.1', :configurations => ['Debug']
diff --git a/Sources/App/iOS/VLCAppDelegate.m b/Sources/App/iOS/VLCAppDelegate.m
index 2c1fc802..21495fd7 100644
--- a/Sources/App/iOS/VLCAppDelegate.m
+++ b/Sources/App/iOS/VLCAppDelegate.m
@@ -19,6 +19,7 @@
 
 #import "VLCAppDelegate.h"
 #import "VLC-Swift.h"
+#import <Sentry/Sentry.h>
 
 @interface VLCAppDelegate ()
 {
@@ -134,6 +135,16 @@
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
+    [SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+        options.dsn = @"https://a92d50327ac74b8b9aa4ea80eccfb267@o447951.ingest.sentry.io/5428557";
+        options.environment = @"integration-tests";
+        options.tracesSampleRate = @1.0;
+        options.enableFileIOTracking = YES;
+        options.attachScreenshot = YES;
+        options.attachViewHierarchy = YES;
+        options.enableUserInteractionTracing = YES;
+    }];
+    
     self.orientationLock = UIInterfaceOrientationMaskPortrait | UIInterfaceOrientationMaskLandscape;
 
     self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
