diff --git a/Podfile b/Podfile
index 6d1a5f78..9a1d70d6 100644
--- a/Podfile
+++ b/Podfile
@@ -30,6 +30,8 @@ pod 'ObjectMapper', git: 'https://github.com/tristanhimmelman/ObjectMapper.git',
 pod 'PromiseKit'
 
 pod 'RealmSwift'
+pod 'Sentry', :git => 'https://github.com/getsentry/sentry-cocoa.git', :commit => '__GITHUB_REVISION_PLACEHOLDER__'
+pod 'SentryPrivate', :git => 'https://github.com/getsentry/sentry-cocoa.git', :commit => '__GITHUB_REVISION_PLACEHOLDER__'
 pod 'UIColor_Hex_Swift'
 pod 'Version'
 pod 'XCGLogger'
@@ -113,7 +115,7 @@ post_install do |installer|
       config.build_settings['WATCHOS_DEPLOYMENT_TARGET'] = '5.0'
       config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
 
-      config.build_settings['SWIFT_INSTALL_OBJC_HEADER'] = 'NO' unless target.name.include? 'Firebase'
+      config.build_settings['SWIFT_INSTALL_OBJC_HEADER'] = 'NO' unless target.name.include? 'Firebase' or target.name.include? 'Sentry'
 
       # disabled arch to stay under the 75 MB limit imposed by apple
       config.build_settings['EXCLUDED_ARCHS[sdk=watchos*]'] = 'arm64'
diff --git a/Sources/App/AppDelegate.swift b/Sources/App/AppDelegate.swift
index edc1510b..ba09893c 100644
--- a/Sources/App/AppDelegate.swift
+++ b/Sources/App/AppDelegate.swift
@@ -13,6 +13,7 @@ import SafariServices
 import Shared
 import UIKit
 import XCGLogger
+import Sentry
 
 let keychain = Constants.Keychain
 
@@ -125,6 +126,14 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         _ application: UIApplication,
         didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
     ) -> Bool {
+        SentrySDK.start { options in
+            options.dsn = "https://6cc9bae94def43cab8444a99e0031c28@o447951.ingest.sentry.io/5428557"
+            options.environment = "integration-tests"
+            options.tracesSampleRate = 1.0
+            options.enableFileIOTracking = true
+            options.profilesSampleRate = 1.0
+        }
+
         if NSClassFromString("XCTest") != nil {
             return true
         }
